tests/testthat/test_reset.R

```diff
@@ -2,12 +2,281 @@ context("reset.R functions")
+
+    test_t_obj  <- t_obj
+    test_t_reset <- resetDGEobj(t_obj)
+
+    # The following are defined in setup.R: t_obj, t_isoform_obj,t_exon_obj, t_protein_obj
+
+    # test gene level data
+    test_gene_obj <- function(gene_data) {
+
+        test_t_obj <- gene_data
+        test_t_reset <- resetDGEobj(test_t_obj)
+
+        # object validation
+        expect_s3_class(test_t_reset, "DGEobj")
+        expect_equivalent(showMeta(test_t_reset), showMeta(test_t_obj))
+
+        # validate level is gene before and after reset
+        test_t_meta <- showMeta(test_t_obj)
+        expect_equal("gene", test_t_meta$Value[3])
+        expect_equal(attr(test_t_obj, "level"), "gene")
+
+        test_t_meta_reset <- showMeta(test_t_reset)
+        expect_equal("gene", test_t_meta_reset$Value[3])
+        expect_equal(attr(test_t_reset, "level"), "gene")
+
+        # reset after subsetting
+        test_t_dim <- dim(test_t_obj)
+        expect_equal(dim(test_t_obj),c(951, 48))
+        test_t_obj <- test_t_obj[c(1:10), ]
+        test_t_reset <- resetDGEobj(test_t_obj)
+        expect_equal(dim(test_t_reset), c(1000, 48))
+
+        # check names after reset
+        expect_named(test_t_obj, c('counts_orig', 'counts', 'design_orig', 'design', 'geneData_orig',
+                                   'geneData', 'granges_orig', 'granges', 'DGEList', 'ReplicateGroupDesign',
+                                   'ReplicateGroupDesign_Elist', 'ReplicateGroupDesign_fit',
+                                   'ReplicateGroupDesign_fit_cm', 'ReplicateGroupDesign_fit_cf',
+                                   'BDL_vs_Sham', 'EXT1024_vs_BDL', 'Nint_vs_BDL', 'Sora_vs_BDL'))
+        expect_named(test_t_reset, c("counts_orig", "counts", "design_orig", "design",
+                                   "geneData_orig", "geneData", "granges_orig", "granges"))
+
+        # testing t_obj with rm item
+        test_t_obj <- rmItem(test_t_obj, "design")
+        expect_false("design" %in% names(test_t_obj))
+        test_t_reset <- resetDGEobj(test_t_obj)
+        expect_true("design" %in% names(test_t_reset)) # check if persists
+
+        # testing t_obj with add item
+        test_t_obj <- addItem(test_t_obj,
+                              item = 'Fred Flintstone',
+                              itemName = 'Cartoon',
+                              itemType = 'meta',
+                              itemAttr = list('MyAttribute' = 'testObject'))
+        expect_true("Cartoon" %in% names(test_t_obj))
+        test_t_reset <- resetDGEobj(test_t_obj)
+        expect_false("Cartoon" %in% names(test_t_reset)) # check if removed
+
+        # testing t_obj after class change
+        test_t_obj <- as.list.DGEobj(test_t_obj)
+        expect_equal(class(test_t_obj), "list")
+        # coerce back into DGEobj, expect reset to work
+        class(test_t_obj) <- "DGEobj"
+        expect_silent(resetDGEobj(test_t_obj))
+    }
+
+    expect_success(test_gene_obj(t_obj))
```

Comment:
Won't be needed once you change these to test_that("description", {<code>}) instead of functions.

Reply:
Note: all the t_ variables are globally set in setup.R so you can just use them as-is in the file anywhere you need.